core/src/main/java/org/wso2/msf4j/interceptor/InterceptorExecutor.java

```diff
@@ -0,0 +1,170 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.msf4j.interceptor;
+
+import org.wso2.msf4j.Request;
+import org.wso2.msf4j.Response;
+import org.wso2.msf4j.exception.InterceptorException;
+import org.wso2.msf4j.interceptor.annotation.RequestInterceptor;
+import org.wso2.msf4j.interceptor.annotation.ResponseInterceptor;
+import org.wso2.msf4j.internal.MicroservicesRegistryImpl;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Class for executing interceptors.
+ */
+public class InterceptorExecutor {
+
+    /**
+     * Process request interceptors.
+     *
+     * @param microServicesRegistry Current micro-services registry
+     * @param method                Method
+     * @param request               msf4j context
+     * @return is request interceptors successful
+     * @throws InterceptorException on interception exception
+     */
+    public boolean processRequestInterceptors(MicroservicesRegistryImpl microServicesRegistry, Method method,
+                                              Request request) throws InterceptorException {
+        Class<?> resourceClass = method.getDeclaringClass();
+        List<Class<?>> subResourceRequestInterceptorClasses = method.isAnnotationPresent(RequestInterceptor.class)
+                ? Arrays.asList(method.getAnnotation(RequestInterceptor.class).value()) : new ArrayList<>();
```

Comment:
Since we don't add new items to `subResourceRequestInterceptorClasses` list, we can use `Collections.emptyList()` instead of creating an new ArrayList.